### ISSUE TITLE
eax: allow variable length nonces

### ISSUE THREAD

**Issue by lopsided98** [RustCrypto/AEADs#239]
Similar to #62, except with EAX. I would like to be able to interoperate with an existing protocol that uses 32-bit nonces. Presumably this could be implemented similarly to #126, but I'm not familiar enough with the codebase to do it easily.

**Comment by Flakebi** [comment]
Is what you need implemented in #231? :)

**Comment by tarcieri** [comment]
I think so. Please reopen if not.

**Comment by lopsided98** [comment]
No, that PR doesn't implement what I need. That allows variable length tags, whereas I need variable length nonces. The nonce size is still hardcoded to the block size: https://github.com/RustCrypto/AEADs/blob/master/eax/src/lib.rs#L192

**Comment by tracktwo** [comment]
I have the same issue with a different protocol: eddystone ETLM uses 48-bit nonce (https://github.com/google/eddystone/blob/master/eddystone-tlm/tlm-encrypted.md). I think the solution used for aesgcm would be enough here too, although it's unfortunate that the order of the generic parameters for nonce and tag sizes would not match the other algorithm without a breaking change.
